### PR TITLE
Introducting Go types

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

Introducing the specification SDK with Go types. See `README.md` for more details.

**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [x] Specification
- [ ] Examples
- [ ] Schema
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] Website
- [x] SDK
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:

**Special notes for reviewers**:

**Additional information (if needed):**

### CODE REVIEW

**Comment by tsurdilo** [comment]
Thanks @ricardozanini !

File: specification/crds/go/go.sum [hunk 1]
```diff
@@ -0,0 +1,371 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (tsurdilo) [inline specification/crds/go/go.sum:1]: Why is this file needed, sorry i don't understand. is it like a in node a package.json but for go?

File: specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
+// +build !ignore_autogenerated
```
Comment (tsurdilo) [inline specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.deepcopy.go:1]: Do we want to add the generated code in the repo too or is some automated build process possible? I'm thinking again like in node where we don't add the generated js typically in the source repo.

File: specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.openapi.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// +build !ignore_autogenerated
```
Comment (tsurdilo) [inline specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.openapi.go:1]: same question about generated code here.

File: specification/crds/go/go.sum (hunk 1 above)
Comment (ricardozanini) [reply specification/crds/go/go.sum:1]: Exactly! It's the dependency management for go.

File: specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.deepcopy.go (hunk 2 above)
Comment (ricardozanini) [reply specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.deepcopy.go:1]: We need this generated code to be able to implement `reflect.DeepEquals` in the object types we are defining here. It's useful for our clients.

File: specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.openapi.go (hunk 3 above)
Comment (ricardozanini) [reply specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.openapi.go:1]: This one generates the boilerplate code for the open api validation, again useful for clients. But this can go away.

File: specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.deepcopy.go (hunk 2 above)
Comment (tsurdilo) [reply specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.deepcopy.go:1]: ok :)

File: specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.openapi.go (hunk 3 above)
Comment (tsurdilo) [reply specification/crds/go/pkg/apis/serverlessworkflow/v1alpha2/zz_generated.openapi.go:1]: already answered in previous comment.

**Comment by ricardozanini** [comment]
Depends on #45

**Comment by ricardozanini** [comment]
@tsurdilo this is just the pure Go part. I'll send another PR once this one get merged with an example project for a custom Kubernetes application using these Go types.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.idea
```
Comment (tsurdilo) [inline .gitignore:1]: Do we need this on the top level ? Maybe put in sdk dir or sdk/go ?

File: sdk/go/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Serverless Specification Go SDK
+
+In this directory you will find all the specification types defined by our schemas in Go.
```
Comment (tsurdilo) [inline sdk/go/README.md:3]: Add link to the schema doc please.

**Comment by tsurdilo** [comment]
@ricardozanini wow..GREAT work!! Thank you! 
Just left some small comments.

File: .gitignore (hunk 4 above)
Comment (ricardozanini) [reply .gitignore:1]: I believe I just edited the one we already have. But yeah, sure, I can add one on `sdk/go` too.
